pinot-common/src/main/java/org/apache/pinot/common/metadata/segment/LLCRealtimeSegmentZKMetadata.java

```diff
@@ -87,11 +87,6 @@ public void setDownloadUrl(String downloadUrl) {
   public ZNRecord toZNRecord() {
     ZNRecord znRecord = super.toZNRecord();
     znRecord.setSimpleField(START_OFFSET, _startOffset);
-    if (_endOffset == null) {
-      // TODO Issue 5359 Keep this until all components have upgraded to a version that can handle _offset being null
```

Comment:
Perhaps we should make a note that before this upgrade, installations should upgrade to at least the release that has https://github.com/apache/incubator-pinot/issues/5359 (Identify the actual release in the PR comments). An installation that has older versions than pr 5359 may break during upgrade.